lib/path.h

```diff
@@ -17,57 +17,12 @@ limitations under the License.
 #ifndef LIB_PATH_H_
 #define LIB_PATH_H_
 
-/*
-  It's 2015 and C++ still does not have a portable way to manipulate pathnames.
-  This code is not portable, but at least the interfaces should be.
-*/
-
-#include "cstring.h"
+#include <filesystem>
 
 namespace Util {
-using namespace P4::literals;
-
 // Represents a filename path, e.g., /usr/local/bin/file.exe
-// FIXME: Can we replace with std::filesystem?
-class PathName final {
- private:
-    static const char pathSeparators[2];
-    cstring str;
-
-    const char *findLastSeparator() const;
-
- public:
-    static inline cstring separator() {
-#ifdef _WIN32
-        return "\\"_cs;
-#else
-        return "/"_cs;
-#endif
-    }
-
-    PathName(cstring str) : str(str) {}             // NOLINT(runtime/explicit)
-    PathName(const char *str) : str(str) {}         // NOLINT(runtime/explicit)
-    PathName(const std::string &str) : str(str) {}  // NOLINT(runtime/explicit)
-    // get the file name extension.  It starts at the last dot.
-    // e.g, exe
-    cstring getExtension() const;
-    // extract just the filename, including the extension
-    // e.g., file.exe
-    PathName getFilename() const;
-    // extract the filename without folder, excluding the extension
-    // e.g., file
-    cstring getBasename() const;
-    // extract the folder
-    // e.g., /usr/local/bin
-    PathName getFolder() const;
-    cstring toString() const { return str; }
-    bool isNullOrEmpty() const { return str.isNullOrEmpty(); }
-    bool operator==(const PathName &other) const { return str == other.str; }
-    bool operator!=(const PathName &other) const { return str != other.str; }
-    PathName join(cstring component) const;
-
-    static PathName empty;
-};
+// We used to have local implementation. Make this alias for now
+using PathName = std::filesystem::path;
```

Comment:
If we make this a separate PR I'd consider replacing this alias entirely. I do not see a reason to keep it, really.

Or, add a deprecation notice, if possible.

Reply:
`std::filesystem::path` is quite long to type, in this sense `PathName` is much shorter even with the namespace. Certainly, this is a bit of bike-shedding :)

But yes, we can get rid of it entirely. Let me split into a separate PR and remove it entirely.